engine/eConstr.ml

```diff
@@ -651,6 +654,10 @@ let substnl subst n c = of_constr (Vars.substnl (List.map to_constr subst) n (to
 let substl subst c = of_constr (Vars.substl (List.map to_constr subst) (to_constr c))
 let subst1 c r = of_constr (Vars.subst1 (to_constr c) (to_constr r))
 
+let substnl_decl subst n d = of_rel_decl (Vars.substnl_decl (List.map to_constr subst) n (to_rel_decl d))
```

Comment:
Indeed, I did not see the trick. So, I applied it also to other functions liable to take benefit of it and I move `sym` in `util.ml` for more generality. For that, I moved casts defined late at earlier places, hoping it is ok for @ppedrot.

Nevertheless, it remains different ways to cast (`cast_rel_decl unsafe_eq` vs `unsafe_to_rel_decl` for instance).